src/lib.rs

```diff
@@ -212,40 +212,46 @@ impl UPTransportZenoh {
     }
 
     fn uattributes_to_attachment(uattributes: &UAttributes) -> anyhow::Result<ZBytes> {
-        let attachment = [
-            UATTRIBUTE_VERSION.to_le_bytes().to_vec(),
-            uattributes.write_to_bytes()?,
-        ];
-        Ok(attachment.iter().collect::<ZBytes>())
+        let mut writer = ZBytes::writer();
+        writer.append(ZBytes::from(UATTRIBUTE_VERSION.to_le_bytes().to_vec()));
+        writer.append(ZBytes::from(uattributes.write_to_bytes()?));
+        let zbytes = writer.finish();
+        Ok(zbytes)
     }
 
     fn attachment_to_uattributes(attachment: &ZBytes) -> anyhow::Result<UAttributes> {
-        let mut attachment_iter = attachment.iter::<Vec<u8>>();
-        // Check the version
-        if let Some(version) = attachment_iter
-            .next()
-            .and_then(Result::ok)
-            .and_then(|v| v.first().copied())
-        {
-            if version != UATTRIBUTE_VERSION {
-                let msg =
-                    format!("UAttributes version is {version} (should be {UATTRIBUTE_VERSION})");
-                error!("{msg}");
-                return Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg).into());
-            }
-        } else {
-            let msg = "Unable to get the UAttributes version".to_string();
+        let mut attachment_iter = attachment.slices();
```

Comment:
I had also started migrating to 1.0.0-rc.2. Here's what I came up with:
```rust
    fn attachment_to_uattributes(attachment: &ZBytes) -> anyhow::Result<UAttributes> {
        if attachment.len() < 2 {
            return Err(UStatus::fail_with_code(
                UCode::INVALID_ARGUMENT,
                "message has no/invalid attachment",
            )
            .into());
        }

        let attachment_bytes = attachment.to_bytes();
        let ver = attachment_bytes[0];
        if ver != UATTRIBUTE_VERSION {
            let msg = format!(
                "Expected UAttributes version {UATTRIBUTE_VERSION} but found version {ver}"
            );
            error!("{msg}");
            return Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg).into());
        }
        // Get the attributes
        let uattributes = UAttributes::parse_from_bytes(&attachment_bytes[1..])?;
        Ok(uattributes)
    }
```

Reply:
Yes, I believe this looks better. But in our original design, we want to use the iterator to get multiple values out orderly.
https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/zenoh.adoc#23-uattribute-mapping
I'm fine with using pure bytes instead of multiple values, which might be simpler for other languages.
If you also agree with that, then we might need to update the spec and other language usage to keep the interoperability.

Reply:
My understanding of the ZBytes type is that the `to_bytes` function simply concatenates the slices of data into one continuous vector, regardless of where the slices had originally been allocated in memory. The outcome should therefore be exactly the same as before, shouldn't it?

Reply:
@sophokles73 Oh, I missed that. Then I think it can work perfectly. Since you've already had another version, do you prefer to create another PR, or it's fine for me to update this PR?
BTW, Zenoh will be released officially today (If nothing goes wrong :laughing: )

Reply:
be my guest 😃 Good luck with the release!

Reply:
looks like the 1.0.0 release has been successful :-) Are you going to update this PR accordingly?

Reply:
Yes, I will update the PR tomorrow to fit the changes!